--- Source/WTF/wtf/StackBounds.cpp.orig	2012-06-17 02:34:04.000000000 +0100
+++ Source/WTF/wtf/StackBounds.cpp	2012-06-17 02:33:46.000000000 +0100
@@ -25,6 +25,9 @@

 #include <mach/task.h>
 #include <mach/thread_act.h>
+#ifndef _DARWIN_C_SOURCE
+#define _DARWIN_C_SOURCE
+#endif
 #include <pthread.h>

 #elif OS(WINDOWS)
--- Source/WTF/wtf/ThreadSpecific.h.orig	2012-06-17 02:41:42.000000000 +0100
+++ Source/WTF/wtf/ThreadSpecific.h	2012-06-17 02:41:48.000000000 +0100
@@ -46,6 +46,15 @@
 #include <wtf/StdLibExtras.h>

 #if USE(PTHREADS)
+
+#if OS(DARWIN)
+#include <mach/task.h>
+#include <mach/thread_act.h>
+#ifndef _DARWIN_C_SOURCE
+#define _DARWIN_C_SOURCE
+#endif
+#endif
+
 #include <pthread.h>
 #elif OS(WINDOWS)
 #include <windows.h>
--- Source/JavaScriptCore/heap/MachineStackMarker.cpp.orig	2012-09-04 01:23:31.000000000 +0100
+++ Source/JavaScriptCore/heap/MachineStackMarker.cpp	2012-09-04 01:24:53.000000000 +0100
@@ -38,6 +38,11 @@
 #include <mach/thread_act.h>
 #include <mach/vm_map.h>
 
+#ifndef _DARWIN_C_SOURCE
+#define _DARWIN_C_SOURCE
+#endif
+#include <pthread.h>
+
 #elif OS(WINDOWS)
 
 #include <windows.h>
--- Source/JavaScriptCore/runtime/JSLock.cpp.orig	2012-09-04 01:25:52.000000000 +0100
+++ Source/JavaScriptCore/runtime/JSLock.cpp	2012-09-04 01:26:31.000000000 +0100
@@ -28,6 +28,13 @@
 #include "ScopeChain.h"
 
 #if USE(PTHREADS)
+
+#if OS(DARWIN)
+#ifndef _DARWIN_C_SOURCE
+#define _DARWIN_C_SOURCE
+#endif
+#endif
+
 #include <pthread.h>
 #endif
 
--- Source/WebCore/loader/FTPDirectoryParser.cpp.orig	2012-09-04 01:27:25.000000000 +0100
+++ Source/WebCore/loader/FTPDirectoryParser.cpp	2012-09-04 01:28:12.000000000 +0100
@@ -29,6 +29,14 @@
 // On Windows, use the threadsafe *_r functions provided by pthread.
 #elif OS(WINDOWS) && (USE(PTHREADS) || HAVE(PTHREAD_H))
 #include <pthread.h>
+
+#elif OS(DARWIN)
+#ifndef _DARWIN_C_SOURCE
+#define _DARWIN_C_SOURCE
+#endif
+
+#include <pthread.h>
+
 #endif
 
 #include <wtf/ASCIICType.h>
--- Source/WTF/wtf/ThreadingPrimitives.h.orig	2012-09-04 01:36:28.000000000 +0100
+++ Source/WTF/wtf/ThreadingPrimitives.h	2012-09-04 01:36:44.000000000 +0100
@@ -43,6 +43,11 @@
 #endif
 
 #if USE(PTHREADS)
+#if OS(DARWIN)
+#ifndef _DARWIN_C_SOURCE
+#define _DARWIN_C_SOURCE
+#endif
+#endif
 #include <pthread.h>
 #endif
 
--- Source/WTF/wtf/InlineASM.h.orig	2012-09-04 01:55:14.000000000 +0100
+++ Source/WTF/wtf/InlineASM.h	2012-09-04 01:55:19.000000000 +0100
@@ -77,7 +77,7 @@
 // FIXME: figure out how this works on all the platforms. I know that
 // on Linux, the preferred form is ".Lstuff" as opposed to "Lstuff".
 // Don't know about any of the others.
-#if PLATFORM(MAC)
+#if OS(DARWIN)
 #define LOCAL_LABEL_STRING(name) "L" #name
 #elif OS(LINUX)
 #define LOCAL_LABEL_STRING(name) ".L" #name
